src/diffusers/pipelines/pyramid_broadcast_utils.py

```diff
@@ -120,40 +32,68 @@ def _enable_pyramid_attention_broadcast(self) -> None:
+                    any(x in name for x in self._pab_temporal_attn_layer_identifiers)
+                    and self._pab_temporal_attn_skip_range is not None
+                    and not module.is_cross_attention
+                )
+                is_cross_attention = (
+                    any(x in name for x in self._pab_cross_attn_layer_identifiers)
+                    and self._pab_cross_attn_skip_range is not None
+                    and module.is_cross_attention
+                )
 
-                skip_range, timestep_range = None, None
-                if module.is_cross_attention and self._pab_cross_attn_skip_range is not None:
-                    skip_range = self._pab_cross_attn_skip_range
-                    timestep_range = self._pab_cross_attn_timestep_range
-                if not module.is_cross_attention and self._pab_spatial_attn_skip_range is not None:
+                if is_spatial_attention:
                     skip_range = self._pab_spatial_attn_skip_range
                     timestep_range = self._pab_spatial_attn_timestep_range
+                if is_temporal_attention:
+                    skip_range = self._pab_temporal_attn_skip_range
+                    timestep_range = self._pab_temporal_attn_timestep_range
+                if is_cross_attention:
+                    skip_range = self._pab_cross_attn_skip_range
+                    timestep_range = self._pab_cross_attn_timestep_range
 
                 if skip_range is None:
                     continue
 
-                module.set_processor(
-                    PyramidAttentionBroadcastAttentionProcessorWrapper(
-                        self, module.processor, skip_range, timestep_range
-                    )
+                # logger.debug(f"Enabling Pyramid Attention Broadcast in layer: {name}")
+                print(f"Enabling Pyramid Attention Broadcast in layer: {name}")
+
+                add_hook_to_module(
+                    module,
+                    PyramidAttentionBroadcastHook(
+                        skip_range=skip_range,
+                        timestep_range=timestep_range,
+                        timestep_callback=self._pyramid_attention_broadcast_timestep_callback,
+                    ),
+                    append=True,
                 )
 
     def _disable_pyramid_attention_broadcast(self) -> None:
         denoiser: nn.Module = self.transformer if hasattr(self, "transformer") else self.unet
         for name, module in denoiser.named_modules():
-            if isinstance(module, Attention) and isinstance(
-                module.processor, PyramidAttentionBroadcastAttentionProcessorWrapper
-            ):
-                logger.debug(f"Disabling Pyramid Attention Broadcast in layer: {name}")
-                module.processor = module.processor._original_processor
+            logger.debug(f"Disabling Pyramid Attention Broadcast in layer: {name}")
+            remove_hook_from_module(module)
```

Comment:
Should there be an option to call `remove_hook_from_module()` with `recurse=True`?

Reply:
In this case, we are iterating in a for-loop the relevant layers to remove hooks from, so recurse would not make sense here